### PR TITLE
Add functional tests for support testing label header

### PR DESCRIPTION
**Pull request by osulzhenko**

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy [hunk 1]
```diff
@@ -2170,6 +2174,112 @@ class CookieSyncSpec extends BaseSpec {
...
+        def uidsCookie = UidsCookie.defaultUidsCookie
+
+        and: "Save account with cookie and privacySandbox configs"
+        def cookieSyncConfig = new AccountCookieSyncConfig(defaultLimit: 1)
+        def privacySandbox = PrivacySandbox.defaultPrivacySandbox
+        def accountAuctionConfig = new AccountAuctionConfig(privacySandbox: privacySandbox)
+        def accountConfig = new AccountConfig(status: AccountStatus.ACTIVE, cookieSync: cookieSyncConfig, auction: accountAuctionConfig)
+        def account = new Account(uuid: accountId, config: accountConfig)
+        accountDao.save(account)
+
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy:2226]: Same empty line

File: src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy [hunk 2]
```diff
@@ -2170,6 +2174,112 @@ class CookieSyncSpec extends BaseSpec {
...
+        when: "PBS processes cookie sync request"
+        def response = prebidServerService.sendCookieSyncRequestRaw(cookieSyncRequest, uidsCookie)
+
+        then:  "Response should contain cookie header"
+        assert response.headers[SET_COOKIE_HEADER] == getCookieDeprecationHeader(privacySandbox.cookieDeprecation.ttlSeconds)
+    }
+
+    def "PBS should set cookie deprecation header from the default account when the accountId from the request is not matched"() {
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy:2261]: when default account contain privacy sandbox

File: src/test/groovy/org/prebid/server/functional/tests/AuctionSpec.groovy [hunk 3]
```diff
@@ -431,4 +440,107 @@ class AuctionSpec extends BaseSpec {
...
+        def bidRequest = BidRequest.defaultBidRequest
+
+        and: "Account in the DB"
+        def accountAuctionConfig = new AccountAuctionConfig(privacySandbox: privacySandbox)
+        def accountConfig = new AccountConfig(status: ACTIVE, auction: accountAuctionConfig)
+        def account = new Account(uuid: bidRequest.accountId, config: accountConfig)
+        accountDao.save(account)
+
+        and: "Sec-Cookie-Deprecation header"
+        def secCookieDeprecation = [(COOKIE_DEPRECATION_HEADER): PBSUtils.randomString]
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/AuctionSpec.groovy:477]: Don't really need set sec-cookie-deprecation header when privacy sandbox is null or false

**Review by marki1an — CHANGES_REQUESTED** [review]
What about this [requirement](https://github.com/prebid/prebid-server/issues/3297#:~:text=If%20the%20call%20to%20/cookie_sync%20doesn%27t%20contain%20an%20account%20ID%2C%20use%20the%20default%20account.%20We%20need%20to%20know%20if%20the%20account%20(or%20default%20account)%20wants%20this%20behavior.) ?

File: src/test/groovy/org/prebid/server/functional/tests/AuctionSpec.groovy [hunk 4]
```diff
@@ -431,4 +439,158 @@ class AuctionSpec extends BaseSpec {
...
+
+        when: "PBS processes auction request with header"
+        defaultPbsService.sendAuctionRequest(bidRequest, secCookieDeprecation)
+
+        then: "BidResponse should have device.ext.cdep from sec-cookie-deprecation header"
```
Comment (AntoxaAntoxic) [inline src/test/groovy/org/prebid/server/functional/tests/AuctionSpec.groovy:460]: probably `BidderRequest should have...`, please check other places

File: src/test/groovy/org/prebid/server/functional/tests/AuctionSpec.groovy [hunk 5]
```diff
@@ -431,4 +439,158 @@ class AuctionSpec extends BaseSpec {
...
+
+        when: "PBS processes auction request with header"
+        pbsService.sendAuctionRequest(bidRequest, secCookieDeprecation)
+
+        then: "BidResponse should have device.ext.cdep from sec-cookie-deprecation header"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.device.ext.cdep == secCookieDeprecation[COOKIE_DEPRECATION_HEADER]
+    }
+
+    def "PBS should set device.ext.cdep from header when default account don't contain privacy sandbox and request account is empty"() {
```
Comment (AntoxaAntoxic) [inline src/test/groovy/org/prebid/server/functional/tests/AuctionSpec.groovy:554]: `should not`

File: src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy [hunk 6]
```diff
@@ -2170,6 +2174,148 @@ class CookieSyncSpec extends BaseSpec {
...
+        def accountAuctionConfig = new AccountAuctionConfig(privacySandbox: PrivacySandbox.getDefaultPrivacySandbox(true, null))
+        def accountConfig = new AccountConfig(status: AccountStatus.ACTIVE, auction: accountAuctionConfig)
+        def account = new Account(uuid: accountId, config: accountConfig)
+        accountDao.save(account)
+
+        when: "PBS processes cookie sync request"
+        def response = prebidServerService.sendCookieSyncRequestRaw(cookieSyncRequest, uidsCookie)
+
+        then:  "Response should contain cookie header"
+        assert extractCookieParameter(response.headers[SET_COOKIE_HEADER], 'Max-Age') == TimeUnit.DAYS.toSeconds(7) as Integer
```
Comment (AntoxaAntoxic) [inline src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy:2276]: I do not see any test that check the whole set-cookie header value

**Review by AntoxaAntoxic — CHANGES_REQUESTED** [review]

**Review by AntoxaAntoxic — APPROVED** [review]

**Review by marki1an — APPROVED** [review]

**Pull request merged**